src/distfile.py

```diff
@@ -8,16 +8,18 @@
 import shutil
 import subprocess
 import urllib
+import urllib.error
 
 from . import config
 from . import util
 
+from typing import Set, Union
 log = logging.getLogger(__name__)
 
 
 class Distfile:
-    def __init__(self, distdir, name, checksum, master_sites, patches,
-                 unsafe_string_sources):
+    def __init__(self, distdir: str, name: str, checksum: str, master_sites: Set[str], patches: Set[str],
+                 unsafe_string_sources: Union[frozenset, Set[str]]) -> None:
```

Comment:
Is the `Union[frozenset, Set[str]]` really necessary? As in, wouldn't `Set[str]` be sufficient?

Reply:
I think `frozenset()` is passed in for this argument somewhere, and `Set[str]` doesn't include frozen sets. `AbstractSet` includes both. I wish the shorter name were the immutable type, but this isn't scala...